### PR TITLE
Added angstrom unit in physics.unit

### PR DESCRIPTION
**Pull request by 1e9abhi1e10**
Angstrom is a unit of length commonly used to express sizes of atoms, molecules or wavelengths of electromagnentic radiation
#### Release Notes

- physics.units
     - Added the unit Angstrom

### CODE REVIEW

**Comment by 1e9abhi1e10** [comment]
When I using `find_out` function in my branch it gives the right result as expected.
```
from sympy.physics import units as u
print(u.find_unit('angstrom'))
```
`Output : ['angstrom', 'angstroms']`
Then why tests failed?

**Comment by 1e9abhi1e10** [comment]
> When I using `find_out` function in my branch it gives the right result as expected.
> 
> ```
> from sympy.physics import units as u
> print(u.find_unit('angstrom'))
> ```
> 
> `Output : ['angstrom', 'angstroms']` Then why tests failed?

fixed!

**Comment by 1e9abhi1e10** [comment]
Is there is a way to write `angstrom` into another single variable because it can't be written in `A` it is already taken by `ampere`.

File: sympy/physics/units/definitions/unit_definitions.py [hunk 1]
```diff
@@ -159,6 +159,8 @@
 mi = mile = miles = Quantity("mile")
 mi.set_global_relative_scale_factor(5280, feet)
 
+angstrom = angstroms = Quantity("angstrom", latex_repr=r'\r{A}')
```
Comment (Upabjojr) [inline sympy/physics/units/definitions/unit_definitions.py:162]: I would recommend to add `.set_global_relative_scale_factor(... , meter)` here because the scale factor relative to meters is the same in all unit systems.

File: sympy/physics/units/systems/si.py [hunk 2]
```diff
@@ -253,6 +253,10 @@
 SI.set_quantity_dimension(astronomical_unit, length)
 SI.set_quantity_scale_factor(astronomical_unit, 149597870691*meter)
 
+SI.set_quantity_dimension(angstrom, length)
+SI.set_quantity_scale_factor(angstrom, 1*meter/10**10)
```
Comment (Upabjojr) [inline sympy/physics/units/systems/si.py:257]: This only sets the scale factor for SI units. The scale factor should be set for all unit systems.

Comment (1e9abhi1e10) [reply sympy/physics/units/systems/si.py:257]: > This only sets the scale factor for SI units. The scale factor should be set for all unit systems.

But when I convert `angstrom` into `inches`or in `foot` or`mm` it gives correct results in my local branch.

Comment (Upabjojr) [reply sympy/physics/units/systems/si.py:257]: `convert_to` uses SI units by default. Try using cgs units.

Comment (1e9abhi1e10) [reply sympy/physics/units/systems/si.py:257]: you are saying to add these things or anything else 
```
SI.set_quantity_scale_factor(angstrom, 1*centimeter/10**8)
SI.set_quantity_scale_factor(angstrom, 1*millimeter/10**7)
SI.set_quantity_scale_factor(angstrom, 3.9370078740157*inch/10**9)
SI.set_quantity_scale_factor(angstrom, 3.280839895*foot/10**10)
```

Comment (Upabjojr) [reply sympy/physics/units/systems/si.py:257]: No. The scale factor should only be defined once, globally for all unit systems.

Comment (1e9abhi1e10) [reply sympy/physics/units/systems/si.py:257]: As you said before that scale factor should be set for all unit systems but now...
Could you pls give an example so that I understand

Comment (Upabjojr) [reply sympy/physics/units/systems/si.py:257]: Only `angstrom.set_global_relative_scale_factor( ... )` should be called. `SI.set_quantity_scale_factor( ... )` only sets the scale factor for SI units.

```python
In [1]: from sympy.physics.units.systems import SI

In [2]: from sympy.physics.units.systems.cgs import cgs_gauss

In [3]: from sympy import *; from sympy.physics.units import *

In [4]: angstrom = angstroms = Quantity("angstrom", latex_repr=r'\r{A}')

In [5]: SI.set_quantity_scale_factor(angstrom, 1*meter/10**10)

In [7]: SI.set_quantity_dimension(angstrom, length)

In [8]: convert_to(angstrom, meter, SI)
Out[8]: meter/10000000000

In [9]: convert_to(angstrom, meter, cgs_gauss)
Out[9]: angstrom
```

As you can see in this example, output 9 does not work, because the scale factor has only been set for SI units, so calling `convert_to` in `cgs_gauss` units does not work. Therefore, you should only use the global set factor in this case (there are other cases where the scale factor is different depending on the chosen unit system, in those cases it's correct to set the quantity scale factor inside the unit system object).

Comment (1e9abhi1e10) [reply sympy/physics/units/systems/si.py:261]: done!

File: sympy/physics/units/systems/cgs.py [hunk 3]
```diff
@@ -66,6 +66,8 @@
 cgs_gauss.set_quantity_scale_factor(ohm, 10**5/speed_of_light**2*second/centimeter)
 cgs_gauss.set_quantity_scale_factor(farad, One/10**5*speed_of_light**2*centimeter)
 cgs_gauss.set_quantity_scale_factor(henry, 10**5/speed_of_light**2/centimeter*second**2)
+cgs_gauss.set_quantity_scale_factor(angstrom, 1*centimeter/10**8)
```
Comment (Upabjojr) [inline sympy/physics/units/systems/cgs.py:69]: Well, this one is not needed because it's already handled by `.set_global_relative_scale_factor( ... )`. The method `.set_quantity_scale_factor( ... )` is supposed to be used for scale factors that depend on the unit system. The scale factor for `angstrom` is the same in all unit systems.

Comment (Upabjojr) [reply sympy/physics/units/systems/cgs.py:69]: Just to be clear:

```python
In [1]: from sympy.physics.units.systems import SI

In [2]: from sympy.physics.units.systems.cgs import cgs_gauss

In [3]: from sympy import *; from sympy.physics.units import *

In [5]: angstrom = angstroms = Quantity("angstrom", latex_repr=r'\r{A}')

In [6]: angstrom.set_global_dimension(length)

In [10]: angstrom.set_global_relative_scale_factor(sympify(1)/10**10, meter)

In [11]: convert_to(angstrom, meter, SI)
Out[11]: meter/10000000000

In [12]: convert_to(angstrom, meter, cgs_gauss)
Out[12]: meter/10000000000

In [13]: convert_to(angstrom, centimeter, cgs_gauss)
Out[13]: centimeter/100000000

In [14]: convert_to(angstrom, centimeter, SI)
Out[14]: centimeter/100000000
```

That is, if you set the scale factor and dimensions to be global, you don't need to specify them for all unit systems.

The scale factor between `angstrom` and other usual length units (meters, centimeters, inches, feet, light years...) is always the same in all unit systems, so it has to be specified as "global".

File: sympy/physics/units/systems/cgs.py [hunk 4]
```diff
@@ -67,6 +67,8 @@
 cgs_gauss.set_quantity_scale_factor(farad, One/10**5*speed_of_light**2*centimeter)
 cgs_gauss.set_quantity_scale_factor(henry, 10**5/speed_of_light**2/centimeter*second**2)
 
+
+
```
Comment (Upabjojr) [inline sympy/physics/units/systems/cgs.py:71]: we usually recommend max 2 whitelines in unindented code (it's not mandatory, but it makes the code cleaner to read).

File: sympy/physics/units/tests/test_unit_system_cgs_gauss.py [hunk 5]
```diff
@@ -31,6 +31,8 @@ def test_conversion_to_from_si():
     assert convert_to(newton, dyne, SI) == 10**5*dyne
     assert convert_to(newton, dyne, cgs_gauss) == 10**5*dyne
 
+def test_angstrom_to_cgs_gauss():
+    assert convert_to(angstrom, centimeter, cgs_gauss) == 1*centimeter/10**8
```
Comment (Upabjojr) [inline sympy/physics/units/tests/test_unit_system_cgs_gauss.py:35]: maybe this can be added to the previous test function (do we need a test function just for angstrom)?

File: sympy/physics/units/tests/test_unit_system_cgs_gauss.py [hunk 6]
```diff
@@ -49,6 +51,6 @@ def test_cgs_gauss_convert_constants():
 
     spc = 25000*second/(22468879468420441*centimeter)
     assert convert_to(ohm, second/centimeter, cgs_gauss) == spc
+    assert convert_to(farad, centimeter, cgs_gauss) == 299792458**2*centimeter/10**5
```
Comment (Upabjojr) [inline sympy/physics/units/tests/test_unit_system_cgs_gauss.py:54]: why was this line changed?

**Comment by Upabjojr** [comment]
is it possible to have all commits squashed into a single commit? This PR has an enormous amount of commits for just a single addition to the units.

**Comment by 1e9abhi1e10** [comment]
> is it possible to have all commits squashed into a single commit? This PR has an enormous amount of commits for just a single addition to the units.

I tried recently but was not able to get desirable things, Should I close this PR and open a new PR for the final review?

**Comment by oscarbenjamin** [comment]
> Should I close this PR and open a new PR for the final review?

If you are not familiar with squashing commits then you might find it easier just to open a new PR and make the final changes in a single commit.

**Comment by 1e9abhi1e10** [comment]
#24630
